service/cziti/sdk.go

```diff
@@ -778,8 +778,23 @@ func zitiContextEvent(ztx C.ziti_context, status C.int, zid *ZIdentity) {
 
 		zid.Name = zid.setNameFromId()
 		zid.Version = zid.setVersionFromId()
+
+		var controllerEvt = dto.ControllerEvent{
+			Fingerprint: zid.Fingerprint,
+			ControllerVersion: zid.Version,
+			IdentityName: zid.Name,
+			ActionEvent: dto.CONTROLLER_CONNECTED,
+		}
+		goapi.BroadcastEvent(controllerEvt)
 		log.Debugf("============ controller connected: %s at %v. MFA: %v", zid.Name, zid.Version, zid.MfaEnabled)
 	} else {
+		var controllerEvt = dto.ControllerEvent{
```

Comment:
imo you should move controllerEvt and the broadcast outside of this if just to reduce code and hopefully avoid any bugs if this code ever changes. just assign the ActionEvent to a var first, and do the create/broadcast after the if